src/app/task/task.service.ts

```diff
@@ -22,4 +22,13 @@ export class TaskService {
     this.allTasksList.unshift(this.createTask(name));
     return this.allTasksList;
   }
+
+  updateTask(id: number, newTask: TaskInterface) {
```

Comment:
in your case it is not id of a task, it's just an index of the array. 
anyway, there is no need in following logic, because you may do like  this.allTasksList[id] = newTask;